tests/functional/authentication/test_auth_allow_with_rebalance.py

```diff
@@ -0,0 +1,155 @@
+        """
+        Mount volume on authenticated client
+
+        Args:
+            mount_obj(obj): Object of GlusterMount class
+        """
+        # Mount volume
+        ret = mount_obj.mount()
+        self.assertTrue(ret, ("Failed to mount %s on client %s" %
+                              (mount_obj.volname,
+                               mount_obj.client_system)))
+        g.log.info("Successfully mounted %s on client %s", mount_obj.volname,
+                   mount_obj.client_system)
+
+        # Verify mount
+        ret = mount_obj.is_mounted()
+        self.assertTrue(ret, ("%s is not mounted on client %s"
+                              % (mount_obj.volname, mount_obj.client_system)))
+        g.log.info("Verified: %s is mounted on client %s",
+                   mount_obj.volname, mount_obj.client_system)
+
+    def _add_brick_rebalance(self):
+        """Create files,Perform Add brick and wait for rebalance to complete"""
+        # Create files on mount point using dd command
+        cmd = ('cd %s;for i in {1..100000};'
+               'do dd if=/dev/urandom bs=1024 count=1 of=file$i;done;'
+               % (self.mounts[0].mountpoint))
+        ret, _, _ = g.run(self.mounts[0].client_system, cmd)
+        self.assertEqual(ret, 0, "Failed to createfiles on mountpoint")
+        g.log.info("Successfully created files on mountpoint")
+
+        # Add brick to volume
+        ret = expand_volume(self.mnode, self.volname, self.servers,
+                            self.all_servers_info)
+        self.assertTrue(ret, "Failed to add brick on volume %s"
+                        % self.volname)
+
+        # Trigger rebalance and wait for it to complete
+        ret, _, _ = rebalance_start(self.mnode, self.volname,
+                                    force=True)
+        self.assertEqual(ret, 0, "Failed to start rebalance on the volume %s"
+                         % self.volname)
+
+        # Wait for rebalance to complete
+        ret = wait_for_rebalance_to_complete(self.mnode, self.volname,
+                                             timeout=1200)
+        self.assertTrue(ret, "Rebalance is not yet complete on the volume "
+                             "%s" % self.volname)
+        g.log.info("Rebalance successfully completed")
+
+    def test_auth_allow_with_rebalance(self):
+        """
+        Validating the FUSE authentication volume options with rebalance
+        Steps:
+        1. Setup and start volume
+        2. Set auth.allow on volume for client1 using ip of client1
+        3. Mount volume on client1.
+        4. Create files on mount point using dd command
+        5. Perform add brick operation
+        6. Trigger rebalance
+        7. Set auth.allow on volume for client1 using hostname of client1.
+        8. Repeat steps from 3 to 7
+        """
+        # Setting authentication on volume for client1 using ip
+        auth_dict = {'all': [self.mounts[0].client_system]}
```

Comment:
Small query, ig client_system will be queried from supplied yaml so are we sure it's always IP here?

Reply:
@leelavg yes we are passing IP in yaml file for clients. This is how all authentication test cases are written for hostname and IP validation..

In case of IP authentication, client_system is being used.
In case of hostname authenticaton we are fetching hostname from server using below -
Eg:
ret, hostname_client2, _ = g.run(self.mounts[1].client_system,
                                         "hostname")

Reply:
> This is how all authentication test cases are written for hostname and IP validation

Ack, i reviewed and figured out `hostname` part :)